formats/notion.ts

```diff
@@ -0,0 +1,103 @@
+import { FormatImporter } from 'format-importer';
+import { ImportResult } from 'main';
+import { FileSystemAdapter, Notice, normalizePath } from 'obsidian';
+import { escapeRegex, fixDuplicateSlashes } from '../util';
+import { cleanDuplicates } from './notion/clean-duplicates';
+import { convertNotesToMd } from './notion/convert-to-md';
+import { copyFiles } from './notion/copy-files';
+import { parseFiles } from './notion/parse-info';
+import { assembleParentIds } from './notion/notion-utils';
+
+export class NotionImporter extends FormatImporter {
+	init() {
+		this.addFolderChooserSetting('Notion HTML export folder', ['html']);
+
+		this.fileLocationSetting?.settingEl.toggle(false);
+		this.folderLocationSetting?.settingEl.toggle(true);
+
+		this.addOutputLocationSetting('Notion');
+	}
+
+	async import(): Promise<void> {
+		let { app, filePaths, folderPaths } = this;
+		let targetFolderPath = (await this.getOutputFolder()).path;
+		// As a convention, all parent folders should end with "/" in this importer.
+		if (!targetFolderPath.endsWith('/')) targetFolderPath += '/';
+		targetFolderPath = fixDuplicateSlashes(targetFolderPath);
+
+		if (filePaths.length === 0) {
+			new Notice('Please pick at least one folder to import.');
+			return;
+		}
+
+		const folderPathsReplacement = new RegExp(
+			`^(${folderPaths
+				.map((folderPath) => escapeRegex('/' + folderPath))
+				.join('|')})`
+		);
+
+		let adapter = app.vault.adapter;
+		if (!(adapter instanceof FileSystemAdapter)) return;
```

Comment:
Any reason why this is needed? I get that we're using readPath, but I think ultimately I'd want to remove the hard requirement on NodeJS.

Reply:
That was vestigial code from when I was starting out. I've removed it.